packages/block-library/src/embed/index.js

```diff
@@ -156,6 +178,47 @@ export function getEmbedEdit( title, icon ) {
 			if ( html || 'photo' === type ) {
 				setAttributes( { type, providerNameSlug } );
 			}
+
+			// If the embedded content is in an iframe with fixed width and height, we
+			// calculate the aspect ratio and set an extra css class so that the rendered
+			// content keeps the correct height no matter how wide the block is set to be.
+			const { height, width } = this.getiFrameHeightWidth( html );
+			if ( undefined !== height && undefined !== width ) {
```

Comment:
I wonder if most of this logic could be moved into `getIframeHeightWidth` and that could become `getAspectRatioClassName` so that `setAttributesFromPreview` isn't doing so much. Or they could be separate functions.